llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -432,12 +432,20 @@ InstructionCost RISCVTTIImpl::getShuffleCost(TTI::ShuffleKind Kind,
     // must be implemented here.
     break;
   case TTI::SK_ExtractSubvector:
+    // Extract at zero is always a subregister extract
+    if (Index == 0)
+      return TTI::TCC_Free;
+
     // Example sequence:
     // vsetivli     zero, 4, e8, mf2, tu, ma (ignored)
     // vslidedown.vi  v8, v9, 2
     return LT.first *
            getRISCVInstructionCost(RISCV::VSLIDEDOWN_VI, LT.second, CostKind);
   case TTI::SK_InsertSubvector:
+    // Insert at zero is always a subregister insert
+    if (Index == 0)
+      return TTI::TCC_Free;
```

Comment:
For inserts we still may need a vslideup if the subvec to insert is < LMUL 1 and the vector isn't undef

Reply:
Good catch.  It doesn't necessarily need to be a slideup, but we do need an instruction to perform the merge, you're right.

Reply:
I reframed this review to be specifically about the extract case.  Once this lands, I'll take another attempt at the insert case.  That ones slightly annoying since I don't see an easy way in the current API to figure out what the passthru is.  I may have to resort to inspecting the operands of the instruction directly.